src/main/java/net/minecraftforge/fml/common/registry/ObjectHolderRef.java

```diff
@@ -44,15 +41,20 @@
 class ObjectHolderRef {
     private Field field;
     private ResourceLocation injectedObject;
-    private boolean isBlock;
-    private boolean isItem;
+    private boolean isValid;
+    private IForgeRegistry registry;
 
 
     ObjectHolderRef(Field field, ResourceLocation injectedObject, boolean extractFromExistingValues)
     {
+        Class<?> type = field.getType();
+        while (IForgeRegistryEntry.class.isAssignableFrom(type) && registry == null) {
+            registry = PersistentRegistryManager.findRegistryByType((Class<IForgeRegistryEntry>)type);
+            type = type.getSuperclass();
```

Comment:
Does the Forge registry system allow for registry types of interfaces?
If so this would need to move to a more stack type search where each class adds its superclass and all interfaces from the class.